### PR TITLE
[refactor](cleanup) Clean up pipeline engine (PART I)

### PR DESCRIPTION
**Pull request by Gabriel39**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

**Review by BiteTheDDDDt — APPROVED** [review]

File: be/src/pipeline/pipeline_fragment_context.cpp [hunk 1]
```diff
@@ -976,31 +1563,100 @@ Status PipelineFragmentContext::send_report(bool done) {
...
+    }
+
+    for (auto& runtime_state : _task_runtime_states) {
+        if (runtime_state->runtime_profile() == nullptr) {
+            continue;
+        }
+
+        auto tmp_load_channel_profile = std::make_shared<TRuntimeProfileTree>();
+
+        runtime_state->runtime_profile()->to_thrift(tmp_load_channel_profile.get());
```
Comment (0AyanamiRei) [inline be/src/pipeline/pipeline_fragment_context.cpp:1653]: In `VNodeChannel::_add_block_success_callback` we use `_state->load_channel_profile()->update(tprofile);`, so here is a empty profile, we can not see any things in profile file about "LoadChannel"

**Pull request merged**